### PR TITLE
Fix issue #214

### LINKED ISSUE #214: Some blocks are not rendered with a volume of size (1140,800,1320)
This is reproducible with both bbic and mem datasources, the volume is cut; only 2 blocks instead 4 seem to be visible at --max-lod=0.

vglrun livre --volume bbic:///gpfs/bbp.cscs.ch/project/proj3/resources/volumes/bbic/allen_average_brain.h5 --max-lod 0
OR
vglrun livre --volume mem:///#1140,800,1320,56 --max-lod 0

screenshot from 2015-11-30 17 56 58

The code in VolumeDataSourcePlugin.cpp -> fillRegularVolumeInfo() might be a good starting point for debugging, but the conflicting documentation in livre/core/data/NodeId.h between RootNode and NodeId constructors complicate the task.

The reported depth of 4 looks suspicious since the LOD structure is the following:

BBIC file v1 - 0 stacks, 1 volumes
Volume v2 [1140, 800, 1320], block size: 64, #blocks (18, 13, 21)
detailed structure:
Level 0: #blocks (18, 13, 21), size (1140, 800, 1320)
Level 1: #blocks (9, 7, 11), size (570, 400, 660)
Level 2: #blocks (5, 4, 6), size (285, 200, 330)
Level 3: #blocks (3, 2, 3), size (142, 100, 165)
Level 4: #blocks (2, 1, 2), size (71, 50, 82)

However increasing the depth to 5 in fillRegularVolumeInfo() breaks the behaviour in multiple places. This bugs needs further investigations as I couldn't find an easy fix.

### PR DESCRIPTION
**Pull request by rdumusc**
I tested the change with 2 different BBIC files, one memory data source and one uvf file, which appeared to rendered as expected

### CODE REVIEW

File: livre/core/data/VolumeDataSourcePlugin.cpp [hunk 1]
```diff
@@ -75,29 +75,29 @@ void VolumeDataSourcePlugin::internalNodeToLODNode(
...
-    const uint32_t zDepth = std::log2( blocksSize[2] );
-
-    const uint32_t depth = std::min( xDepth, std::min( yDepth, zDepth ));
-    blocksSize = blocksSize / ( 1u << depth );
-
-    info.rootNode = RootNode( depth + 1, blocksSize );
+                               info.voxels[2] ) * info.worldSpacePerVoxel;
+
+    // Create the rootNode of the LOD hierarchy
+    const Vector3ui blockSize = info.maximumBlockSize - info.overlap * 2;
```
Comment (bilgili) [inline livre/core/data/VolumeDataSourcePlugin.cpp]: const Vector3f blockSize = info.maximumBlockSize - info.overlap * 2;
 const Vector3f voxels = info.voxels;
 const Vector3ui numBlocks =  ( info.voxels / blockSize ) + 1u;

Comment (rdumusc) [inline livre/core/data/VolumeDataSourcePlugin.cpp]: As discussed offline, the ceil operation is correct in terms of boundary conditions, this is not. I'm writing a unit to remove all doubts.

**Comment by rdumusc** [comment]
@bilgili Unit test added

**Comment by rdumusc** [comment]
retest this please

File: tests/core/volumeInformation.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    BOOST_CHECK_EQUAL( volume.rootNode.getBlockSize(), livre::Vector3ui( 1u ));
+    BOOST_CHECK_EQUAL( volume.worldSpacePerVoxel, 1.f / 2048.f );
+    BOOST_CHECK_EQUAL( volume.worldSize, livre::Vector3f( 1.f,
+                                                          2037.f / 2048.f,
+                                                          1.f ));
+}
+
+BOOST_AUTO_TEST_CASE( nonOctreeLODTree )
+{
+    // This highlights a know corner case, where the parent-child relationship
```
Comment (bilgili) [inline tests/core/volumeInformation.cpp]: a known

Comment (rdumusc) [inline tests/core/volumeInformation.cpp]: done :-)

**Comment by bilgili** [comment]
+1

**Comment by rdumusc** [comment]
retest this please

File: livre/core/data/VolumeDataSourcePlugin.cpp [hunk 3]
```diff
@@ -75,29 +75,29 @@ void VolumeDataSourcePlugin::internalNodeToLODNode(
...
+    const Vector3ui numBlocks(
+                std::ceil( float( info.voxels.x( )) / blockSize.x( )),
+                std::ceil( float( info.voxels.y( )) / blockSize.y( )),
+                std::ceil( float( info.voxels.z( )) / blockSize.z( ))
```
Comment (eile) [inline livre/core/data/VolumeDataSourcePlugin.cpp]: Does it make sense to add a vmml::Vector::ceil() so this becomes ```const Vector3ui numBlocks( Vector3f( info.voxels.x / blockSize).ceil( ))```?

Comment (rdumusc) [inline livre/core/data/VolumeDataSourcePlugin.cpp]: In general this could be a useful addition. How should it be done? in place, with a const & return, or return a copy? I know that in Qt they typically provide two functions, like ceil() which does it in place and ceiled() which returns a copy.
Also, it wouldn't be enough to simplify much the next two operation which apply a log2 and >> operation on the vector elements, so I didn't jump into it.

Comment (bilgili) [inline livre/core/data/VolumeDataSourcePlugin.cpp]: or a templated free function, like the std
vmml::ceil( ... )

Comment (eile) [inline livre/core/data/VolumeDataSourcePlugin.cpp]: Or a partial template specialization for std::ceil.

Comment (bilgili) [inline livre/core/data/VolumeDataSourcePlugin.cpp]: +1 Next week I will have a look at that.

Comment (rdumusc) [inline livre/core/data/VolumeDataSourcePlugin.cpp]: OK sounds good!
